### ISSUE TITLE
Underscore: chained reduce returns wrong type

### ISSUE THREAD

**Issue by palortoff** [DefinitelyTyped/DefinitelyTyped#7931]
A chained reduce on an array returns a _ChainSingle<TResult>, event if TResult is an array. In this case the reduce result is fully chainable and therefore should be a _Chain<TResult>.

Example:

```
_.chain([1, 2, 3])
            .reduce((acc, x) => [x], [])
            .map(() => 1)
            .value()
```

Results in TS error:

```
error TS2339: Property 'map' does not exist on type '_ChainSingle<any[]>'
```

**Comment by vvakame** [comment]
please send a pull request. I'll review it.

**Comment by reubenrybnik** [comment]
This was actually fixed by #45304, I just wasn't familiar with the magic of the word "fix" in a PR so I used a later PR in the same set to close this out.
